CHANGELOG.md

```diff
@@ -1,5 +1,75 @@
 # CHANGELOG
 
+## 2.15.0rc0 (2024-07-22)
+
+We are excited to announce the release candidate for MLflow 2.15.0 (`mlflow-2.15.0rc0`). This release candidate provides users with an early preview of the major features and changes coming in the stable release. We encourage you to try out these new features, and if you encounter any issues or have suggestions, please open an issue on our GitHub repository!
+
+
+### Try It Out
+
+To install the release candidate, run the following command:
+
+```
+pip install mlflow==2.15.0rc0
+```
+
+Please note that this is a release candidate and may contain bugs or incomplete features. We encourage you to test it in a non-production environment and provide feedback on any issues you encounter.
+
+### Major features:
+
+- **LlamaIndex Flavor** - MLflow now introduces a native integration with [LlamaIndex](https://www.llamaindex.ai/), one of the most popular libraries for building GenAI apps centered around custom data. This integration allows you to log LlamaIndex indices within MLflow, load and deploy for inference with different engine types. MLflow also provides comprehensive tracing support for LlamaIndex operations, offering unprecedented transparency into complex queries. Check out [MLflow LlamaIndex documentation](https://mlflow.org/docs/latest/llms/llama-index/index.html) to get started! (#12633, @michael-berk, @B-Step62)
+
+- **OpenAI Tracing** - We've enhanced our OpenAI integration with a new tracing feature that works seamlessly with MLflow OpenAI autologging. Users can now enable tracing of their OpenAI API usage with a single `mlflow.openai.autolog()` call, then MLflow automatically logs valuable metadata such as token usage, providing deeper insights into your OpenAI-powered applications. To start exploring this new capability, please check out [the tracing documentation](https://mlflow.org/docs/latest/llms/tracing/index.html#automatic-tracing)! (#12267, @gabrielfu)
+
+- **Enhanced Model Deployment with New Validation Feature** - To improve the reliability of model deployments, MLflow has added a new method to validate your model before deploying it to the inference endpoint. This feature helps eliminate typical errors in input and output handling, streamlining the process of model deployment and increasing confidence in your deployed models. By catching potential issues early, you can ensure a smoother transition from development to production. (#12710, @serena-ruan)
+
+- **Custom Metrics Definition Recording for Evaluations** - We've strengthened the flexibility of defining custom metrics for model evaluation by automatically logging and versioning metrics definitions, including models used as judges and prompt templates. With this new capability, you can ensure reproducibility of evaluations across different runs and easily reuse evaluation setups for consistency, facilitating more meaningful comparisons between different models or versions. (#12487, #12509, @xq-yin)
```

Comment:
@serena-ruan @xq-yin Do we have any docs/examples to add here for prompting users to try these features out?

Reply:
+1 we should add a quick entry to the docs for this feature.

Reply:
agree. the docs fell off my plate. let me send a PR to update the doc today. but don't think this PR is dependent on the docs update

Reply:
Thanks! This PR will be out as a part of the stable release not now (as discussed this morning) so I can include the link.

Reply:
@B-Step62 oh I found we have this already: https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai_metric#mlflow.metrics.genai.retrieve_custom_metrics guess it's created by docstring?

Reply:
Great, that works for the quick link. Ideally we can add some entry in https://mlflow.org/docs/latest/llms/llm-evaluate/index.html because those python api doc has low visibility, but not urgent.